### PR TITLE
fix(gatsby): Properly slugify fields with periods

### LINKED ISSUE #26776: file system routing: . in slug string breaks route
## Description

Found another bug in file routing. You use slugify for cretaing slugs from strings. For some reason it is not removing "."s. So while slugify would output "mrs.-doubtfire" somewhere else the actual slug gets shortened to "mrs"

### Steps to reproduce

include a "." in your slug field

### Expected result

should remove . from slug

### Actual result

doesn't remove period then url slug gets truncated.

### PR DESCRIPTION
**Pull request by blainekasten**
## Description

Fixes #26776

### CODE REVIEW

File: packages/gatsby-plugin-page-creator/src/derive-path.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export function derivePath(
     const replaceSlashesValue = (nodeValue + ``).replace(/\//g, `(REPLACED)`)
     const slugifiedWithoutSlashesValue = slugify(replaceSlashesValue, {
       lower: true,
+      remove: /\./,
```
Comment (pvdz) [inline packages/gatsby-plugin-page-creator/src/derive-path.ts:54]: Would this not need the global flag? (`/\./g`)

**Review by pvdz — APPROVED** [review]

File: packages/gatsby-plugin-page-creator/src/derive-path.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export function derivePath(
     const replaceSlashesValue = (nodeValue + ``).replace(/\//g, `(REPLACED)`)
     const slugifiedWithoutSlashesValue = slugify(replaceSlashesValue, {
```
Comment (wardpeet) [inline packages/gatsby-plugin-page-creator/src/derive-path.ts:52]: @blainekasten could we use [@sindresorhus/slugify](https://npm.im/@sindresorhus/slugify) instead? It does what we need without any settings.

https://repl.it/@WardPeeters/slugify-package
image

Comment (blainekasten) [reply packages/gatsby-plugin-page-creator/src/derive-path.ts]: Yeah, this looks much better! Switching over

File: packages/gatsby-plugin-page-creator/src/derive-path.ts (hunk 1 above)
Comment (blainekasten) [reply packages/gatsby-plugin-page-creator/src/derive-path.ts]: Good catch on the review, turns out I didn't need this. I tested, but I also changed libs so this code is gone

**Review by wardpeet — APPROVED** [review]
Awesome! Looks good to me

**Pull request merged**